javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,219 @@
+
+let nameArray = [];
+let todoArray = [];
+
+function getReply(command)
+{
+    let wordString = command.split(" "); 
+    let KeyWord = wordString[0];  // extract the keyword
+    let nameString;
+    let nameString2;
+
+    switch(KeyWord){
+        //--------------------------------------------------------------//
+        case 'Hello':
+            nameString = wordString[4]; // extract the name (as it is on 4th index)
+            if(nameString == '')
+            {
+                console.log('please enter your name');
+            }
+            else
+            {    
+                // compare with name array and add if not exist or skip if its already exist
+                if(nameArray.includes(nameString.toLowerCase()) )
+                {
+                    console.log('Name already exists, so skipping it');
+                }
+                else
+                {
+                    console.log('name do not exists, so adding to the list');
+                    nameArray.push(nameString.toLowerCase()); // add to the array
+                    console.log(`Nice to meet you:  ${nameString} !`);
+                    //console.log(`Total Names in the array ${nameArray.length}`);  
+                }
+        
+            }
+            break;
+
+        //--------------------------------------------------------------
+
+            case 'Add':    
+                
+                if(wordString.length > 4)
+                {
+                    nameString2 = wordString;
+                    nameString2.splice(0, 1); // remove first element
+                    nameString2.splice(-1, 1); // remove last element
+                    nameString2.splice(-1, 1); // remove 2nd last element, whih is infact last now
```

Comment:
Or you can do:
`todoTask.splice(-2)` which will delete all elements from the second to last element.